tests/data/discrete_curves_data.py

```diff
@@ -0,0 +1,637 @@
+            DiscreteCurves, self.space_args_list, self.n_vecs_list
+        )
+
+
+class L2CurvesMetricTestData(_RiemannianMetricTestData):
+
+    ambient_manifolds_list = [r2, r3]
+    metric_args_list = [
+        (ambient_manifolds,) for ambient_manifolds in ambient_manifolds_list
+    ]
+    shape_list = [(10, 2), (10, 3)]
+    space_list = [
+        DiscreteCurves(ambient_manifolds)
+        for ambient_manifolds in ambient_manifolds_list
+    ]
+    n_points_list = random.sample(range(2, 5), 2)
+    n_tangent_vecs_list = random.sample(range(2, 5), 2)
+    n_points_a_list = random.sample(range(2, 5), 2)
+    n_points_b_list = [1]
+    batch_size_list = random.sample(range(2, 5), 2)
+    alpha_list = [1] * 2
+    n_rungs_list = [1] * 2
+    scheme_list = ["pole"] * 2
+
+    def exp_shape_test_data(self):
+        return self._exp_shape_test_data(
+            self.metric_args_list, self.space_list, self.shape_list
+        )
+
+    def log_shape_test_data(self):
+        return self._log_shape_test_data(self.metric_args_list, self.space_list)
+
+    def squared_dist_is_symmetric_test_data(self):
+        return self._squared_dist_is_symmetric_test_data(
+            self.metric_args_list,
+            self.space_list,
+            self.n_points_a_list,
+            self.n_points_b_list,
+            atol=gs.atol * 1000,
+        )
+
+    def exp_belongs_test_data(self):
+        return self._exp_belongs_test_data(
+            self.metric_args_list,
+            self.space_list,
+            self.shape_list,
+            self.n_tangent_vecs_list,
+            belongs_atol=gs.atol * 1000,
+        )
+
+    def log_is_tangent_test_data(self):
+        return self._log_is_tangent_test_data(
+            self.metric_args_list,
+            self.space_list,
+            self.n_points_list,
+            is_tangent_atol=gs.atol * 1000,
+        )
+
+    def geodesic_ivp_belongs_test_data(self):
+        return self._geodesic_ivp_belongs_test_data(
+            self.metric_args_list,
+            self.space_list,
+            self.shape_list,
+            self.n_points_list,
+            belongs_atol=gs.atol * 1000,
+        )
+
+    def geodesic_bvp_belongs_test_data(self):
+        return self._geodesic_bvp_belongs_test_data(
+            self.metric_args_list,
+            self.space_list,
+            self.n_points_list,
+            belongs_atol=gs.atol * 1000,
+        )
+
+    def exp_after_log_test_data(self):
+        return self._exp_after_log_test_data(
+            self.metric_args_list,
+            self.space_list,
+            self.n_points_list,
+            rtol=gs.rtol * 100,
+            atol=gs.atol * 10000,
+        )
+
+    def log_after_exp_test_data(self):
+        return self._log_after_exp_test_data(
+            self.metric_args_list,
+            self.space_list,
+            self.shape_list,
+            self.n_tangent_vecs_list,
+            rtol=gs.rtol * 100,
+            atol=gs.atol * 10000,
```

Comment:
Do you know what is going on with exp and log here?

For future work: the tangent vectors generated to test log_after_exp should be generated with their norm being less than the injectivity radius of the metric defining the exp. 

Nicolas put the injectivity radii of many metrics, e.g. here for the hypersphere:
https://github.com/geomstats/geomstats/blob/bc82fbfb5aae9dbb44e2bd99ddfd5fe9f6a427ed/geomstats/geometry/hypersphere.py#L1088

Thus, log_after_exp_test_data could be adapted in the parent class.